app/services/foreman_bootdisk/iso_generator.rb

```diff
@@ -85,8 +85,13 @@ def self.generate(opts = {})
         EOT
       end
 
-      # Temporary directory cannot be cleaned inprocess due to send_file offloaded to web server
-      wd = Dir.mktmpdir('bootdisk-iso-')
+      # Temporary directory cannot be cleaned in-process due to send_file offloaded to web server and
+      # the call is asynchronous. Also $TMPDIR (/tmp or /var/tmp) cannot be used because Foreman runs
+      # with PrivateTmp systemd setting. And the plugin does not ship its own systemd-tmpfiles config,
+      # so let's delete old temporary files explicitly for files older than 30 minutes:
+      Dir.glob(Rails.root.join('tmp/bootdisk-iso-*')).select{|f| File.ctime(f) < (Time.now - (60*30)) }.each{|f| FileUtils.rm_f(f)}
```

Comment:
I think this also works since Rails.root is a [Pathname](https://ruby-doc.org/stdlib-2.6.3/libdoc/pathname/rdoc/Pathname.html):
```suggestion
      Rails.root.glob('tmp/bootdisk-iso-*').select{|f| File.ctime(f) < (Time.now - (60*30)) }.each{|f| FileUtils.rm_f(f)}
```